src/sentry/static/sentry/app/api.tsx

```diff
@@ -58,8 +84,29 @@ export function paramsToQueryArgs(params) {
   return p;
 }
 
+// TODO: move this somewhere
+type APIRequestMethod = 'POST' | 'GET' | 'DELETE' | 'PUT';
+
+type FunctionCallback<Args extends any[] = any[]> = (...args: Args) => void;
+
+type RequestCallbacks = {
+  success?: (data: any, textStatus?: string, xhr?: JQueryXHR) => void;
+  complete?: (jqXHR: JQueryXHR, textStatus: string) => void;
+  error?: FunctionCallback;
```

Comment:
I'm unable to type this properly based on its usage from this file alone. So I'm leaving it as `FunctionCallback`.

When Sentry is significantly migrated to TS, we can revisit and refine this type.

Reply:
Should we be marking things like this somehow?